### PR TITLE
[EXPERIMENT][WIP] Add SelectiveSSM/PagedSelectiveSSM ops for Mamba2 hybrid models

### PR DESCRIPTION
**Pull request by mlukasze**
> ⚠️ AUTOMATICALLY GENERATED BY OMEGA AGENT — REQUIRES HUMAN REVIEW ⚠️
> This PR was created by an AI agent as part of automated model enablement.
> A human maintainer must review and approve it before it can be considered for merge.
> Do **NOT** merge without human review and sign-off.

## Summary

This draft adds two new internal OpenVINO ops for Mamba2-style hybrid decoder models:
- `SelectiveSSM` (stateful)
- `PagedSelectiveSSM` (paged / continuous-batching form)

Implemented scope in this PR:
- Core op definitions, shape inference, visitor/type-prop tests
- Reference math + template backend support
- Stateful and paged fusion passes
- CPU backend + targeted CPU tests
- GPU backend + targeted GPU unit coverage
- Internal op docs/spec pages

### Common stack
- `ov_core_unit_tests --gtest_filter='type_prop.selective_ssm_*:type_prop.paged_selective_ssm_*:attributes.selective_ssm_*:attributes.paged_selective_ssm_*'`
- `ov_transformations_tests --gtest_filter='TransformationTests.SelectiveSSMFusion_*:TransformationTests.PagedSelectiveSSMFusion_*'`
- `ov_template_func_tests --gtest_filter='*SelectiveSSM*'`

### CPU
- built `openvino_intel_cpu_plugin` and `ov_cpu_func_tests`
- `ov_cpu_func_tests --gtest_filter='smoke_SelectiveSSM/*:smoke_PagedSelectiveSSM/*'`
- result: 8 passed / 2 skipped (paged f16/bf16 cases skipped by existing ISA guards on this host)

### GPU
- built `openvino_intel_gpu_plugin`
- built `ov_gpu_unit_tests`
- `ov_gpu_unit_tests --gtest_filter='smoke_selective_ssm_gpu_test/*:smoke_paged_selective_ssm_gpu_test/*'`
- result: 8 passed
- includes larger SelectiveSSM cases up to Granite-like `num_heads=64, head_dim=64, state_size=128`

### Exporter divergence
The current local `optimum-intel` branch for Granite exports a newer fully-vectorized selective-scan formulation with no `Loop`, so it does **not** match the reference loop-based fusion contract used for this PR.

For real-IR fusion validation I fetched `openvino-agent/optimum-intel@nemotron_3`, which contains the loop-based `SelectiveSSMRecurrentCell -> Loop` exporter path referenced during design.

### Real Granite fusion verification
Using `ibm-granite/granite-4.0-h-micro` exported from `openvino-agent/optimum-intel@nemotron_3`:
- serialized IR contained `Loop: 36`
- CPU compiled exec graph showed `SelectiveSSM: 36`
- GPU compiled exec graph also showed `SelectiveSSM: 36`

So fusion is confirmed on a real Granite export for both CPU and GPU plugin compile paths.

### Remaining blocker
GPU e2e inference on the fused real Granite export is still not numerically healthy:
- CPU fused model is sane enough for the probe prompt (`argmax(logits)` matched PyTorch)
- GPU fused model compiles and contains `SelectiveSSM: 36`, but inference on the real Granite export still produces NaNs
- focused follow-up debug localized the failure to GPU `SelectiveSSM` execution itself on real Granite shapes/values: the first fused recurrence layer is already badly wrong at prompt length 1 (CPU output/state max-abs `0.2546/0.2961` vs GPU `194.375/20.40625`)
- the tensors feeding that first fused layer (`dA`, `dtB`, `x`, `C_expanded`, zero init-state) still match CPU closely, and the full-model GPU logits first become NaN at prompt length 2

This means the remaining gap is **not** missing fusion, missing primitive registration, or the exporter-path divergence described above; it is a GPU real-model numerical/runtime integration issue still needing follow-up.

## Round 2 update (focused follow-up, time-boxed)

No code fix landed in this round; no new commits pushed. Investigation significantly narrowed
the root cause of the GPU real-model numerical bug described above:

- **Layout/format hypothesis: refuted.** Instrumented the GPU kernel dispatch to confirm `x`,
  `initial_state`, `output`, `output_state` are `bfyx` (planar) in both the passing GPU unit test
  (including the Granite-matching `num_heads=64/head_dim=64/state_size=128/f16` case) and the real
  Granite export's compiled graph.
- **Key new finding:** feeding the exact real tensor values (captured from a real CPU run of the
  first Granite `SelectiveSSM` layer) into a minimal standalone single-op model reproduces the
  CPU result almost exactly on GPU (diffs ~1e-5, normal fp32 rounding). But the same op, with the
  same real inputs (verified to match closely between CPU/GPU, diffs 5e-4 to 4e-3), embedded in
  the full 36-layer/~700-node real network, diverges massively on GPU only (CPU output max-abs
  `0.20` vs GPU `104.19`). This rules out the kernel math, layout handling, and basic fp16/fp32
  precision as the cause — the bug is graph-scale-specific (likely GPU memory-pool/command-queue
  scheduling interaction for this custom stateful multi-output primitive at real-network scale).
- **Output buffer-pool aliasing hypothesis: refuted.** Forced `can_share_buffer(false)` on the
  `selective_ssm` node (mirroring `cldnn::data`/`mutable_data`'s idiom for un-pooled buffers) and
  re-ran the same comparison: no change at all.

See `agent-results/enable-operator/summary.md` ("Round 2") and `result.json` in
`openvinotoolkit/omega` for full detail, including recommended next steps (bisecting the model by
layer count, GPU command-queue/event tracing, and checking per-layer stateful-Variable memory
allocation) for whoever picks this up next.

## Deferred / follow-up work
- GPU real-model numerical issue on fused Granite export
- Full `ov_gpu_func_tests` target is still blocked in this environment by unrelated pre-existing OpenCL header issues in existing tests
- Paged / continuous-batching runtime wiring in GenAI remains out of scope here
- CPU JIT kernel and GPU optimized kernel remain Phase 2 work; this PR carries reference-only functional implementations

## References
- op spec reference: `agent-results/enable-operator/reference/pr36372_spec.diff`
- POC reference: `agent-results/enable-operator/reference/pr36412_poc.diff`
- exporter reference: `agent-results/enable-operator/reference/pr1789_optimum.diff`
- tracking issue: openvinotoolkit/omega#79

### AI Assistance
- *AI assistance used: yes*
- *If yes, summarize how AI was used and what human validation was performed (build/tests/manual checks).* This PR was generated by the OMEGA enable-operator agent. Local validation performed by the agent included OpenVINO build/test runs, targeted CPU and GPU test execution, real Granite export attempts, and compiled-model/runtime inspection for fusion verification.

### CODE REVIEW

**Comment by mlukasze** [comment]
productized in separated PRs
